### PR TITLE
Adds code coverage to FormatStringDialog

### PR DESCRIPTION
**Pull request by ricardobossan**
Related #10773

## Proposed changes

- Adds code coverage for `FormatStringDialog`.

## Customer Impact

- None

## Regression?

- No

## Risk

- Minimal

## Test methodology

- Unit tests

## Test environment(s)

- 9.0.100-rc.1.24452.12
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12185)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        action.Should().NotThrow();
+    }
+
+    [Fact]
+    public void Dirty_ReturnsDefault() => _formatStringDialog.Dirty.Should().Be(false);
+
+    [Fact]
+    public void SetsListControl_DoesNotThrow()
+    {
+        Action action = () => _formatStringDialog.ListControl = new ListBox();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs:59]: Please make sure to dispose this ListBox

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    [Fact]
+    public void FormatControlFinishedLoading_AdjustsButtonPositionsCorrectly()
+    {
+        var okButtonField = typeof(FormatStringDialog).GetField("_okButton", BindingFlags.NonPublic | BindingFlags.Instance);
+        var cancelButtonField = typeof(FormatStringDialog).GetField("_cancelButton", BindingFlags.NonPublic | BindingFlags.Instance);
+        var formatControlField = typeof(FormatStringDialog).GetField("_formatControl1", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        var okButton = new Button { Top = 10, Left = 10 };
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs:78]: These 3 controls should be disposed

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    [Fact]
+    public void FormatControlFinishedLoading_AdjustsButtonPositionsCorrectly()
+    {
+        var okButtonField = typeof(FormatStringDialog).GetField("_okButton", BindingFlags.NonPublic | BindingFlags.Instance);
+        var cancelButtonField = typeof(FormatStringDialog).GetField("_cancelButton", BindingFlags.NonPublic | BindingFlags.Instance);
+        var formatControlField = typeof(FormatStringDialog).GetField("_formatControl1", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        using Button okButton = new() { Top = 10, Left = 10 };
+        using Button cancelButton = new() { Top = 10, Left = 30 };
+        using FormatControl formatControl = new() ;
```
Comment (Epica3055) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs:84]:
```suggestion
        using FormatControl formatControl = new();
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    {
+        Action action = FormatStringDialog.End;
+
+        action.Should().NotThrow();
+    }
+
+    [Fact]
+    public void FormatControlFinishedLoading_AdjustsButtonPositionsCorrectly()
+    {
+        var okButtonField = typeof(FormatStringDialog).GetField("_okButton", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (LeafShi1) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/FormatStringDialogTests.cs:78]: Please try use TestAccessor().Dynamic, like
```
 Mock<ITypeDescriptorContext> mockContext = new(MockBehavior.Strict);
 FormatStringDialog formatStringDialog =new(mockContext.Object);
 var okButtonField = formatStringDialog.TestAccessor().Dynamic._okButton;
```

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**